stm32H743 nucleo board cannot flash after tests/drivers/flash

**Describe the bug**
build the tests/drivers/flash
flash to the nucleo_h743zi  target  
flashing is done and test runs but a error msg output at the end of the west flash
After that, the next flash (of same or another binary) fails always fails :

```
-- west flash: rebuilding
ninja: no work to do.
-- west flash: using runner openocd
-- runners.openocd: Flashing file: ./build/zephyr/zephyr.hex
Open On-Chip Debugger 0.11.0+dev-00242-g7036ed509-dirty (2021-09-20-14:06)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
init
Info : clock speed 1800 kHz
Info : STLINK V3J2M1 (API v3) VID:PID 0483:374E
Info : Target voltage: 3.282024
Error: init mode failed (unable to connect to the target)
in procedure 'init'

FATAL ERROR: command exited with status 1: $HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/bin/openocd -s ./boards/arm/nucleo_h743zi/support -s $HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts -f ./boards/arm/nucleo_h743zi/support/openocd.cfg '-c init' '-c targets' -c 'reset halt' -c 'flash write_image erase ./build/zephyr/zephyr.hex' -c 'reset halt' -c 'verify_image ./build/zephyr/zephyr.hex' -c 'reset run' -c shutdown
```

The third one is correct.


Please also mention any information which could help others to understand
the problem you're facing:
 - Zephyr OS 20c1e1a8000b3da7817f670825178391a9649016
 - SDK 0.13.1


**To Reproduce**
Steps to reproduce the behavior:
1. west build -p auto -b nucleo_h743zi samples/hello_world/
2. west flash  --> flash & run = OK
3. west build -p auto -b nucleo_h743zi tests/drivers/flash
4. west flash  --> flash & run but "FATAL ERROR: command exited with status 1"
5. west build -p auto -b nucleo_h743zi samples/hello_world/
6. west flash -> no flash + 
```
Error: init mode failed (unable to connect to the target)
in procedure 'init'

FATAL ERROR: command exited with status 1:
```
7. west flash  --> flash & run = OK

**Expected behavior**
flash and run on each command

**Logs and console output**

```
-- west flash: rebuilding
ninja: no work to do.
-- west flash: using runner openocd
-- runners.openocd: Flashing file: $HOME/zephyrproject/zephyr/build/zephyr/zephyr.hex
Open On-Chip Debugger 0.11.0+dev-00242-g7036ed509-dirty (2021-09-20-14:06)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
init
Info : clock speed 1800 kHz
Info : STLINK V3J2M1 (API v3) VID:PID 0483:374E
Info : Target voltage: 3.282024
Info : stm32h7x.cpu0: hardware has 8 breakpoints, 4 watchpoints
Error: mem2array: Read @ 0x5c001004, w=4, cnt=1, failed
Error executing event examine-end on target stm32h7x.cpu0:
$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg:236: Error: 
in procedure 'init' 
in procedure 'stm32h7x_dbgmcu_mmw' called at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 168
in procedure 'stm32h7x_mmw' called at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 260
in procedure 'stm32h7x_mrw' called at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 242
at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 236
Info : starting gdb server for stm32h7x.cpu0 on 3333
Error: couldn't bind gdb to socket on port 3333: Address already in use
Error: mem2array: Read @ 0x5c001004, w=4, cnt=1, failed
Error executing event examine-end on target stm32h7x.cpu0:
/$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg:236: Error: 
in procedure 'init' 
in procedure 'ocd_process_reset' 
in procedure 'ocd_process_reset_inner' called at file "embedded:startup.tcl", line 260
in procedure 'stm32h7x_dbgmcu_mmw' called at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 168
in procedure 'stm32h7x_mmw' called at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 260
in procedure 'stm32h7x_mrw' called at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 242
at file "$HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/stm32h7x.cfg", line 236
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x0800127c msp: 0x24000700
    TargetName         Type       Endian TapName            State       
--  ------------------ ---------- ------ ------------------ ------------
 0* stm32h7x.cpu0      hla_target little stm32h7x.cpu       halted

target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x0800127c msp: 0x24000700
Info : Device: STM32H74x/75x
Info : flash size probed value 2048
Info : STM32H7 flash has dual banks
Info : Bank (0) size is 1024 kb, base address is 0x08000000
Info : Padding image section 0 at 0x0800754c with 20 bytes (bank write end alignment)
Warn : Adding extra erase range, 0x08007560 .. 0x0801ffff
auto erase enabled
wrote 30048 bytes from file ./build/zephyr/zephyr.hex in 1.256025s (23.362 KiB/s)

target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x0800189c msp: 0x24000dc0
verified 30028 bytes in 0.070764s (414.395 KiB/s)


FATAL ERROR: command exited with status 1: $HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/bin/openocd -s $HOME/zephyrproject/zephyr/boards/arm/nucleo_h743zi/support -s $HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts -f ./boards/arm/nucleo_h743zi/support/openocd.cfg '-c init' '-c targets' -c 'reset halt' -c 'flash write_image erase ./build/zephyr/zephyr.hex' -c 'reset halt' -c 'verify_image ./build/zephyr/zephyr.hex' -c 'reset run' -c shutdown
```

Using twister, the testcase fails if the flash mass erased (like the stm32Cubeprogrammer) is not performed _before_ running.
```
FATAL ERROR: command exited with status 1: $HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/bin/openocd -s $HOME/zephyrproject/zephyr/boards/arm/nucleo_h743zi/support -s $HOME/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts -f $HOME/zephyrproject/zephyr/boards/arm/nucleo_h743zi/support/openocd.cfg '-c init' '-c targets' -c 'reset halt' -c 'flash write_image erase ./zephyr/twister-out/nucleo_h743zi/tests/drivers/flash/drivers.flash.stm32/zephyr/zephyr.hex' -c 'reset halt' -c 'verify_image ./twister-out/nucleo_h743zi/tests/drivers/flash/drivers.flash.stm32/zephyr/zephyr.hex' -c 'reset run' -c shutdown
make[3]: *** [zephyr/cmake/flash/CMakeFiles/flash.dir/build.make:72: zephyr/cmake/flash/CMakeFiles/flash] Error 1
make[2]: *** [CMakeFiles/Makefile2:3409: zephyr/cmake/flash/CMakeFiles/flash.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:3416: zephyr/cmake/flash/CMakeFiles/flash.dir/rule] Error 2
make: *** [Makefile:748: flash] Error 2
```

--- a/boards/arm/nucleo_h743zi/board.cmake
+++ b/boards/arm/nucleo_h743zi/board.cmake
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
+board_runner_args(openocd --cmd-post-verify "reset halt")
 board_runner_args(jlink "--device=STM32H743ZI" "--speed=4000")
 board_runner_args(stm32cubeprogrammer "--port=swd" "--reset=hw")
 

